src/DetailsView/components/load-assessment-button.tsx

```diff
@@ -3,31 +3,125 @@
 
 import { AssessmentDataParser } from 'common/assessment-data-parser';
 import { InsightsCommandButton } from 'common/components/controls/insights-command-button';
+import { Tab } from 'common/itab';
+import {
+    AssessmentStoreData,
+    PersistedTabInfo,
+} from 'common/types/store-data/assessment-result-data';
+import { TabStoreData } from 'common/types/store-data/tab-store-data';
+import { VersionedAssessmentData } from 'common/types/versioned-assessment-data';
+import { UrlParser } from 'common/url-parser';
 import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
-import { LoadAssessmentHelper } from 'DetailsView/components/load-assessment-helper';
+import { LoadAssessmentDialog } from 'DetailsView/components/load-assessment-dialog';
 import * as React from 'react';
 
 export type LoadAssessmentButtonDeps = {
     detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
     assessmentDataParser: AssessmentDataParser;
-    loadAssessmentHelper: LoadAssessmentHelper;
+    urlParser: UrlParser;
+    fileReader: FileReader;
+    document: Document;
 };
 export interface LoadAssessmentButtonProps {
     deps: LoadAssessmentButtonDeps;
+    tabStoreData: TabStoreData;
+    assessmentStoreData: AssessmentStoreData;
+}
+export interface LoadAssessmentButtonState {
+    newTab: Tab;
+    prevTab: PersistedTabInfo;
```

Comment:
Here and ChangeAssessmentDialog: we should not use "PersistedTabInfo" unless we need to.

Reply:
As far as I can tell, this has not been fixed.

We should use the Tab interface in these places if that suffices. The PersistedTabInfo interface links with our store data/app knowledge and we should limit components that have it, unless necessary.

Reply:
Don't know how I missed that the first time around, sorry about that! I've implemented this change now.